extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/user/UserService.java

```diff
@@ -88,6 +88,7 @@ private void putAllUsers(Map<String, User> users, LDAPConnection ldapConnection,
             // Set search limits
             LDAPSearchConstraints constraints = new LDAPSearchConstraints();
             constraints.setMaxResults(confService.getMaxResults());
+            constraints.setDereference(confService.getDereferenceAliases());
```

Comment:
I think that approach probably makes sense.  I think this setMaxResults was a recent-ish change to the code to allow searching on large LDAP trees and avoid whatever built-in default existed in the JLDAP code, but I could be wrong about that.  Anyway, probably makes sense to move the constraints to a more global area and apply them universally.  I'll see what kind of progress I can make on that...though any specific help/suggestions would be welcome.